tests/test_gp_var_dist.py

```diff
@@ -39,149 +39,62 @@ def test_toeplitz_GP_lat_prior(use_fast_toeplitz=True):
     print('Y:', Y.shape, np.std(Y), np.quantile(Y, 0.99))
 
     data = torch.tensor(Y, device=device, dtype=torch.get_default_dtype())
-    # specify manifold, kernel and rdist
-    manif = mgp.manifolds.Euclid(m, dfit)
-    #kernel = mgp.kernels.Linear(n, dfit)
-    kernel = mgp.kernels.Linear(n,
-                                dfit,
-                                ard=True,
-                                learn_scale=False,
-                                Y=Y,
-                                Poisson=Poisson)
-
-    print("use fast toeplitz?", use_fast_toeplitz)
-    lat_dist = mgp.rdist.EP_GP(manif,
-                               m,
-                               n_samples,
-                               torch.Tensor(ts),
-                               Y=Y,
-                               initialization='fa',
-                               use_fast_toeplitz=use_fast_toeplitz)
-    _scale = torch.ones(n_samples, d, m) * .2 * (1 + torch.randn(m) / 100
-                                                )  #n_diag x T
-    lat_dist._scale = nn.Parameter(data=inv_softplus(_scale),
-                                   requires_grad=True)
-
-    ###construct prior
-    lprior = mgp.lpriors.Null(manif)
-
-    # generate model
-    if Poisson:
-        likelihood = mgp.likelihoods.NegativeBinomial(n, Y=Y)
-        #likelihood = mgp.likelihoods.Poisson(n)
-    else:
-        likelihood = mgp.likelihoods.Gaussian(n, Y=Y, d=dfit)
-    z = manif.inducing_points(n, n_z)
-    mod = mgp.models.SvgpLvm(n, m, n_samples, z, kernel, likelihood, lat_dist,
-                             lprior).to(device)
-
-    print(mod.lat_dist.name)
-    ### test that training runs ###
-    n_mc = 16
-
-    def cb(mod, i, loss):
-        if i % 5 == 0:
-            print('')
-        return
-
-    mgp.optimisers.svgp.fit(data,
-                            mod,
-                            optimizer=optim.Adam,
-                            n_mc=n_mc,
-                            max_steps=5,
-                            burnin=50,
-                            lrate=10e-2,
-                            print_every=5,
-                            stop=cb,
-                            analytic_kl=True)
-
-    try:
-        print('lat ell, scale:',
-              mod.lat_dist.f.ell.detach().flatten(),
-              mod.lat_dist.f.scale.detach().flatten())
-        print('prior:', mod.lprior.ell.detach().flatten())
-    except:
+
+    for GP in [mgp.rdist.GP_diag, mgp.rdist.GP_circ]:
```

Comment:
You should maybe add something that prints what iteration of this loop we are in, otherwise it'll be hard to debug which model is failing if it fails